### PR TITLE
ModelExperimental: Allow changing custom shaders at runtime

### PR DESCRIPTION
**Pull request by ptrgags**
This PR allows custom shaders to be swapped out at runtime. This is similar to how styling is applied in #9896. These two methods of styling are considered mutually exclusive.

[Local Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=7VVZTxsxEP4ro+3LRgpOOKrSEOgRQEICKpW0UtWtKq89SVy89spHuMR/7+xujg0gtY99aKKsM/fM99neEXoVC3ZyW6JTBZrA9SnyEB16hobnGi+sRN22wyEEF/EgM5mZcwdzhTfoSGvwBkZNvq+1Ls0SUcsjS4HKoMuSziou1xGvZlw+jR1FH2zRWNKHzAB9tJrOgjLTupnB0vO8rWVfLs/Pxt3Gf+L4tOq2yTLG2zCA740JoNeDkdXWQVAaPeR3IJUP3AiEYCHMEAQv0PGlf5bMrZKrnBc0SXq6EM5MGQNMfL12QRlLorgvfhZVTxc8EGwEWbH408mS7jrtw4ZUPZd+TKrJJHokZOYodtM+63ehfmyzPkH4pzg2pcitRVuMh+BUHgNxWlqvgrLmZMTuoVdlw73NdI8r6Qf7ZWnULMmIsqRTax/X/DmU/+n7O/q2V/T1/xn6avQwPCGvXnaPx42x4Y8Av/w2gjwqLYkuvwq9UWFWAXbsuLAgbFHSteGpvSoqujXTZ9Z8Rm+jE8gmzhYfPIWfyfT1m7f7/f2mNdHaN4PW5ZCZuunmlmFeoEFW0l1EOMwJES5luuin5XZvbTG2bUP1veJGCu6DxipsbKdTjR9jCLZCaTSzlhjjUHGAjjYH+KaBpAsTrj3SEo2o4IdUzFBco+xAjdCiDmvPQMAunOBd66y0RzuoZ0u6ydCHO41HSxbfq6K0LlQQpoz1Ahalph3ie3mkfFTG+2qkxvlVsFbnROjDUgOQc3E9dTYaOQA3zXm6t9OF5a/P9tfRACVBQawOYK+8balz66jDLcelin7T+PistKqPUauBObqgBNdbnI6+GUChpNT4vOpWsOUAdjYqL025JWaKTeui9LDXBmwo1RyUPHzhbQNCc+/JMolaX6l7zJKjYY/8n4Vqy6uqn6hzze8qt9n20XmjZIwNeyS+HLkA4Unm3w)

@sanjeetsuhag could you review?

### CODE REVIEW

**Comment by sanjeetsuhag** [comment]
@ptrgags Updated the PR with the fix. The problem was that the `CPUStylingPipelineStage` was not being applied because the tiles already loaded had a custom shader at the time of initialization. That check has been removed, now  `CPUStylingPipelineStage` runs everytime.

**Comment by ptrgags** [comment]
@sanjeetsuhag thanks for looking into this! code changes look reasonable, let's see if CI passes

File: Source/Scene/ModelExperimental/ModelExperimental3DTileContent.js [hunk 1]
```diff
@@ -173,6 +173,7 @@ ModelExperimental3DTileContent.prototype.update = function (
   model.colorBlendAmount = tileset.colorBlendAmount;
```
Comment (lilleyse) [inline Source/Scene/ModelExperimental/ModelExperimental3DTileContent.js:173]: Commenting on code outside the diff...

I don't think we want `ModelExperimental3DTileContent.prototype.applyStyle` to set the style directly. I think it should continue to call `applyStyle`. Also the `ModelExperimental` style setter should early exit if it sees the same style. I think this leads to the more consistent behavior. Say someone writes an application that sets `model.style = myStyle` every frame, we wouldn't want to regenerate commands every frame.

Comment (lilleyse) [reply Source/Scene/ModelExperimental/ModelExperimental3DTileContent.js:173]: I pushed a commit with this change

File: Source/Scene/ModelExperimental/ModelExperimental.js [hunk 2]
```diff
@@ -866,3 +851,23 @@ ModelExperimental.prototype.applyColorAndShow = function (style) {
...
+ModelExperimental.prototype.applyStyle = function (style) {
+  // The style is only set by the ModelFeatureTable. If there are no features,
+  // the color and show from the style are directly applied.
+  if (
+    defined(this.featureTableId) &&
+    this.featureTables[this.featureTableId].featuresLength > 0
+  ) {
+    var featureTable = this.featureTables[this.featureTableId];
+    featureTable.applyStyle(style);
+    this._color = undefined;
```
[the remark this answers is not in the document]
Comment (lilleyse) [reply Source/Scene/ModelExperimental/ModelExperimental.js:869]: I don't think we have to handle this in the PR, but something to consider for later

**Comment by sanjeetsuhag** [comment]
Tested the following Sandcastles:
- [Switching between `style` and `customShader` for 3D Tiles](http://localhost:8080/Apps/Sandcastle/index.html#c=7VVtb9M6FP4rR+EDqdS5HQzd0ZXpQgFp0saVWEFCZLpy4pPW4NiRX7oXtP/OcdK0aQeCj3zAVeL4vPn4eY5PZ+hkqNibmxqtrFB7rt4i98GiY6h5rvDCCFR9PbwAbwOeZDrTK25hJfEaLUk1XsOsjfexkaVZUjTrmSFHqdFmyWDjl6uAl0su9n1nwXlTtZr0W6aBhpKLpZd60SQz6SzP+1L24d352XzY2peWL2K2bZQ53vgJfG5VAKMRzIwyFrxU6CC/BSGd57pA8Ab8EqHgFVre2WfJykixiXlBJ0nfrhdnug4eStfMQ5Da0LK4q/6vYk4X3BNsBFm1/hhkyXAb9tvOKr47OyZkWQaHhMwKi6fpmI2H0LwO2Zgg/JUfW5DnwTotxr23Mg+eOK2Nk14a/WbG7mAUo+HRbrj7zeqKfTF01CzJiLJk0Ejvt/xZFH/p+z36Djf0jf8Y+hr00O+R10xPX89bZcsfAf7u0wzyIJUgutzG9Vr6ZQTsteWFgcJUNbUNR+lFr2C3TJ8Z/R6dCbZAVlpTvXTkfibSZ/88Px4ft6kVvbqZ9JpDppuk2y7DXIEaWU29iHBYESJciHSdT8/szphqbvqK+LvkWhTceYXRbW4WC4WvgvcmojRbGkOMcYgcoKXiANcmkAyh5MohTUEXEX5IiyUWX1EMoEEoPrLcl8axzoD1T0eQBy2wpIYoTqJhZ+T8rcKfE3IZ1ekmNEEWr8EEepIo06IpEde7Mu34nCWxccfzUGOOvuljusSPY1UeU2lfDfv2V9vFfffZMEETIMHR7ruf++7JfgrAlt82Yotiw3UyTKZNtNNu239lVRvrY0mljI08VrWiG+NGeSC4KbZzg5PO+JE3RuVU4D1Ycl58XVhDuU3ALnKeHj0ZQvfEw59sbWsqDaryCRzVNz1xbiwle2C5kMHtKu8fbC2bttJLYIXWy4KrA06tUE+gkkIofLjrgTf1BJ7s7NypckOVWu1q11tPR33ApkKuQIoXP/j3hUJx50hTBqUu5R0Vw+l0RPYPXJXhcdf/KHPFb6PZ8vD0vBUyxqYjWv7Ycw3CXuTv)
- [Switching between `style` and `customShader` for Model](http://localhost:8080/Apps/Sandcastle/index.html#c=hVVtb+M2DP4rgvfhHCCR0zb3srRXDE1frkC7HZZs+3AuBtmmHe1kyZDkNO2h/32U/BIn7bAgiSWKpMiHD+kFGF6X9GpbgeYlSMvENTBbazAUJEsE3KsMxPCcfCZW13Aay1humCYbDo+gUSrhkSwaf396WRgHqd8vFBpyCToORqeNVaUMt1xJtGttFkxbXDF5QnOtyksoNIAJY0nI5Oj4hE4/zmYfjn4eO8FsRqfvpycfpx/8dhrLzu+6OgjlC7CMy+Irt+n6dyVEOB0T9+0Mcr6F7FqzElaaSZMrXe5i6kWGCpUiNl5PXfc2N4BZMau0jzMOpNJ2HQfjZvcIxsbBLrhE1LBcs+wQrkVtrCqbk/CHsyVE8GJtMXCP/7zTvBtK6R+/3t2uxo1+rlnhCtR4WcHWzsm35oiQKCILJZQmlgswJHkiGTeWyRSIVcSugaSYjGadfhxsFM96n/dYvPC63dzKqrYkN/45Jlwq3KbP5d+li+meWWQKsqRsF6MWjcbtj72d++/0aMbzvDaAyGwgPQmn1FXJ/R1RV6z/s6MFWk7asCizVvOkRkLRjmlXC/pMIucNZvvuXvrdA/1HYapYM2R3MPLSl1FPdZ+hC9DTm5oUq08rbAy8YINXsSzzPGiL9ap1PLFvhM27Ihe4nmME0bKC1ESXzLLIW5nIqd0pV8voQm1vZVOuLCrE6jpK1HbCOxF1XvoUuipovp2/QeP1QTsMyRx2kHSQ9RhhV/Xr1ueVELwyyBL6183y06w/fqOfmqNRq5IOyD4ftESDddMsPgmqMdanrwgZx/IiSWWY1zL1UyP0GiPicWzL0VCY5uJppS5ULV2ey2oNGhptmuwJx6QtQlZjB6PTOdLtfV/xtuyxXDKZpcxYAa6+K1UUAi5qa5XjyWKtFHKWEcdCdCotMU02wZjkTBi8Zhd0uob0O2Rt2O7H80NpX0M6BApJh7FDjkM0O3VqjYqxTwIOZol/nFyusNOX7jjs3SL2bgjMyUDiZAiKi88MBkbz+RYHbtS7XHCUO9vwnYbsnWvMT9jYD+Oh/sNu89ItPYr4IEhpaO7dj3w/q/9IfUeSxluDnq9RMA7OvK/z7spfeFnhFCa1FiGlkYWyEjgrTJTUCLOlqTGj0075J6uUSLC1B5AkLP1eaMeVOdFFwsLZ8Zh0P5f46U63Qkogo+ZkVm0H4kRpDHaisddqs3/48upq7gfqIIANaMvxbTNh+BJAWpY8ywS8vnViVTUnx3s3d0eJQoaW+6ft1WfRELCzjG8Izz6/8a4mqWDG4EleC7Hkz0iE87MI9V+ZCuWnym8YuWBPTm19dH7XCCmlZxFu37ZsQTjw/C8)
- [Switching between `style` and `color` on a Model](http://localhost:8080/Apps/Sandcastle/index.html#c=bVRtb9s2EP4rhPahMmBTTuK2mesGQ5wmK9Z2wWxsH6Z+oMSTRJQiBZJynA357z2KkiMnNixI9/7wuTuuwYq2pp/2DRhRg3JM3gJzrQFLQbFMwlfNQY7t5CNxpoUPqUrVjhmyE/AABrUKHsg65Pu708VplHfyWmOgUGDSaPIhRDXaCie0wrg+Zs2Mwy+mLmhhdH0DpQGwcaoImZ2dX9D5+8Xi3dmvU69YLOj87fzi/fxdJ85TNeStmhdQfgfGhSrvhcurv7SU8XxK/H8IKMQe+K1hNWwNU7bQpn7GdFBZKnWO3HR++vYQcwd4Kua06XCmkdLGVWk0DdIDWJdGAVyoVns2MX8gjdoc42mD1CIZO+Sccd5l6uu/Ir+j5k66Iv7fuxFS4vcSSyWbBnKb3DDHki7KJt7ti2YcTHKt95+VdUzlwJNSbm+TTO9nYlBRn6VHTQLGr8wZsV+eIKJ6QeiYjjhkIIf2TgcF9uXw3ef8JKVorBac/nO3uVwczCc6EkyT3iXXUpsDtLWX6P3nb39469Mk0N0dghrE+niPlAkL1FWg4qJVeTd3cecxIR2PfTtyrGkYLeTjVl/rVvlzbpoKDARvmh0pp6RvAm9xBjDpkszp24ACITyFtqdqwxTPmXUSfH+3uiwlXLfOaYUbsq60tkAY4aIoMKlyxFa+adgPUjBpscwz6LyC/AfwHrZ/RPFSG6Ba9yjheBXC6+JmKyRsvHkYohGpI43X4Wl9Ybsk/44NBMXI3wIeJG65j43fZLKFN7ha9Oxykkbfp+OI78/C0/DZEYQvgtMKp7Aj2VDgvcGDXzhyR2w0jVad19WQ7DdRN7h8pDUypjRxUDeS4YWSZC1y42hubSjnf784rWWG+zg6bsbyH6XxDV4SU2YsXpxPyfDM6eVzNI439hHHYEkWzX6kzrTBxs0MLkhrj41Pr0oL1bRuDGAHxgm8ZGZMihJnqRacS3hddeZ0syTnR5UHU6ZxrOpja196lYwJW3GxI4J/PHFFk1wya9FStFJuxH/Y5KtVgv6vQqXuroI/Eblkj96tOrv6EpSU0lWC4unInoQXmX8C)

File: Source/Scene/ModelExperimental/ModelExperimental.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sanjeetsuhag) [reply Source/Scene/ModelExperimental/ModelExperimental.js]: Fixed.

**Comment by ptrgags** [comment]
alright, CI is passing so I'm merging. Thanks @sanjeetsuhag and @lilleyse for the help with debugging this evening!

**Pull request merged**